frontend/src/app/HashRouterElement.tsx

```diff
@@ -54,33 +47,21 @@ const hashRouteMapper = [
 
 export const HashRouterElement = ({
   element,
-  strict,
-  redirectTo = LOGIN_ROUTE,
+  strict = false,
 }: HashRouterElementProps): React.ReactElement => {
   const location = useLocation()
-  const state = location.state as { from: Location } | undefined
-  const { isAuthenticated } = useAuth()
 
   // Retire this custom routing after July 2024
   if (location.hash.startsWith('#!/')) {
     // angular routes that need to be mapped
     for (const { regex, getTarget } of hashRouteMapper) {
       const match = location.hash.match(regex)
       if (match) {
-        redirectTo = getTarget(match as FormRegExpMatchArray)
+        const redirectTo = getTarget(match as FormRegExpMatchArray)
         return <Navigate replace to={redirectTo} state={{ from: location }} />
       }
     }
   }
 
-  if (isAuthenticated && strict) {
-    return <Navigate to={state?.from.pathname ?? ROOT_ROUTE} replace />
-  }
-
-  return (
-    <Flex flexDir="column" height="100vh" pos="relative">
-      <GovtMasthead />
-      {element}
-    </Flex>
-  )
+  return <PublicElement element={element} strict={strict} />
```

Comment:
Right! So when HashRouter was first introduced the `ROOT_ROUTE` was `/` and was private so `HashRouter` replicated the code of `PrivateElement`.

With the route change that this landing PR introduces, the landing zone is `/` and is public. I initially changed `HashRouter` to now replicate `PublicElement` instead if Private (in [this commit](https://github.com/opengovsg/FormSG/pull/3857/commits/6c4411220d470b404fe4313d7a88c90f437c5e79)), but I figured `HashRouter` should return the `PublicElement` instead of duplicating its code. So this is what [this latest commit](https://github.com/opengovsg/FormSG/pull/3857/commits/455fa9a4f9843ad33d5c5f2a9dc515c962f765d6) is doing. Should all work 🤞 

Some things might still warrant some discussions:
* It's a bit funny that the routes are now defined like this:
```javascript
export const LANDING_ROUTE = '/'
export const ROOT_ROUTE = '/workspace'
```
 Perhaps we should refactor the names like this:
```javascript
export const ROOT_ROUTE = '/'
export const ADMIN_ROUTE = '/workspace'
```
(just a suggestion)

* I was now a little confused about the `/workspace` url namespace 😅 . Are we now using `/workspace/admin/forms` or just `/workspace//forms`? The redirection rules in HashRouter will need to correctly reflect whatever final decision we take.

Reply:
/workspace is its own route, once an admin goes into their form it becomes /admin/forms/:formId

i see where the confusion is, yeah, should rename `ROOT_ROUTE` for sure... maybe the workspace page can be `/admin` too, though i think `/workspace` works better.